compiler-core/src/build/package_compiler.rs

```diff
@@ -104,7 +106,7 @@ where
         stale_modules: &mut StaleTracker,
         incomplete_modules: &mut HashSet<EcoString>,
         telemetry: &dyn Telemetry,
-    ) -> Outcome<Vec<Module>, Error> {
+    ) -> Outcome<(Vec<Module>, Vec<type_::ModuleInterface>), Error> {
```

Comment:
Prefer structs to tuples so it's clear what they are

Reply:
Should I call the struct `Compiled`, `CompileResult` or something else?